src/BenchmarkDotNet/Engines/CacheClearingStrategies/MemoryAllocator.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace BenchmarkDotNet.Engines.CacheClearingStrategies
+{
+    internal class MemoryAllocator : IMemoryAllocator
+    {
+        public void AllocateMemory()
+        {
+            const int howManyMb = 14;
+            const int howManyPass = 10;
+            const int sizeOfOneArray = 64;
+
+            for (int i = 0; i < (howManyMb * 1024 * 1024 / sizeOfOneArray) * howManyPass; i++)
+            {
+                var tmpArray = new int[sizeOfOneArray];
```

Comment:
Do we really have to allocate memory before each iteration? I guess it will be enough to preallocate a huge array in advance and reuse it.

Reply:
@AndreyAkinshin Thanks for review! What do you mean by "preallocate a huge array"? Should I create a private array:
```c#
private readonly int[] memory = new int[howManyMb* 1024 * 1024]
```
And then use it instead of allocate a new one?

Reply:
Yes, we should create this array only once.

Reply:
I've pushed changes.